Nombre: Emilio Rico Hernández
Clase: 1
Challenge: Explorador de CSV
Fecha: 2026-09-25

# Challenge 1 — Explorador de CSV

Para este challenge usé `ventas_1.csv`, las transacciones de una tienda de tecnología (`Fecha`, `Producto`, `Region`, `Vendedor`, `Cantidad`, `Precio`, `Ventas`). La idea es hacer una primera exploración con Pandas para entender qué hay en el archivo antes de sacar conclusiones grandes.

## Parte 1 — Carga de datos

In [1]:
import pandas as pd

df = pd.read_csv("ventas_1.csv")

In [2]:
# Primeras 5 filas
df.head()

,Fecha,Producto,Region,Vendedor,Cantidad,Precio,Ventas
0,2026-01-01,Teclado,Este,Diego,1,800.25,800.25
1,2026-01-01,Mouse,Sur,Paula,3,438.48,1315.44
2,2026-01-01,Audifonos,Sur,Carlos,3,608.42,1825.26
3,2026-01-01,Mouse,Norte,Jorge,5,449.33,2246.65
4,2026-01-02,Mouse,Sur,Carlos,4,445.68,1782.72


In [3]:
# Últimas 5 filas
df.tail()

,Fecha,Producto,Region,Vendedor,Cantidad,Precio,Ventas
1010,2026-08-10,Monitor,Sur,Paula,1,4012.58,4012.58
1011,2026-08-10,Monitor,Centro,Ana,2,4075.93,8151.86
1012,2026-08-10,Impresora,Centro,Ana,3,3127.70,9383.10
1013,2026-08-10,Monitor,Sur,Carlos,1,3967.42,3967.42
1014,2026-08-10,Monitor,Sur,Carlos,1,4103.54,4103.54


In [4]:
# Número de filas y columnas
print("Filas y columnas:", df.shape)

Filas y columnas: (1015, 7)


In [5]:
# Nombres de columnas
df.columns

Index(['Fecha', 'Producto', 'Region', 'Vendedor', 'Cantidad', 'Precio',
       'Ventas'],
      dtype='str')

In [6]:
# Tipos de datos
df.dtypes

Fecha           str
Producto        str
Region          str
Vendedor        str
Cantidad      int64
Precio      float64
Ventas      float64
dtype: object

## Parte 2 — Exploración

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1015 entries, 0 to 1014
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Fecha     1015 non-null   str    
 1   Producto  1015 non-null   str    
 2   Region    1015 non-null   str    
 3   Vendedor  1015 non-null   str    
 4   Cantidad  1015 non-null   int64  
 5   Precio    1015 non-null   float64
 6   Ventas    1015 non-null   float64
dtypes: float64(2), int64(1), str(4)
memory usage: 55.6 KB


In [8]:
df.describe()

,Cantidad,Precio,Ventas
count,1015.000000,1015.000000,1015.000000
mean,2.509360,4090.802532,7462.978759
std,1.390182,5301.863364,9516.019997
min,1.000000,418.500000,426.280000
25%,1.000000,587.665000,1717.805000
50%,2.000000,862.240000,3342.840000
75%,3.000000,4241.150000,8606.680000
max,6.000000,16112.310000,48336.930000


In [9]:
# Cuántos valores distintos tiene cada columna
df.nunique()

Fecha        222
Producto       6
Region         4
Vendedor       8
Cantidad       6
Precio      1007
Ventas      1013
dtype: int64

Con `info()`, `describe()` y `nunique()` veo que no hay valores nulos: las 7 columnas tienen 1,015 valores. Hay 6 productos, 4 regiones y 8 vendedores, y las ventas cubren 222 días distintos. `Cantidad` va de 1 a 6 piezas, pero `Precio` va de $418 a $16,112, un rango enorme para un solo archivo.

## Parte 3 — Selección

In [10]:
# Una columna
df["Vendedor"].head()

0     Diego
1     Paula
2    Carlos
3     Jorge
4    Carlos
Name: Vendedor, dtype: str

In [11]:
# Dos o más columnas
df[["Fecha", "Producto", "Ventas"]].head()

,Fecha,Producto,Ventas
0,2026-01-01,Teclado,800.25
1,2026-01-01,Mouse,1315.44
2,2026-01-01,Audifonos,1825.26
3,2026-01-01,Mouse,2246.65
4,2026-01-02,Mouse,1782.72


In [12]:
# Un subconjunto de filas (de la 100 a la 104)
df.iloc[100:105]

,Fecha,Producto,Region,Vendedor,Cantidad,Precio,Ventas
100,2026-01-23,Audifonos,Sur,Carlos,3,595.44,1786.32
101,2026-01-23,Mouse,Sur,Paula,2,451.34,902.68
102,2026-01-23,Monitor,Centro,Ana,2,4313.12,8626.24
103,2026-01-23,Mouse,Este,Luis,5,447.42,2237.10
104,2026-01-23,Teclado,Norte,Jorge,5,856.58,4282.90


In [13]:
# Filas que cumplen una condición: ventas de más de $10,000
grandes = df[df["Ventas"] > 10000]
print("Ventas mayores a $10,000:", len(grandes), "de", len(df))
grandes.head()

Ventas mayores a $10,000: 220 de 1015


,Fecha,Producto,Region,Vendedor,Cantidad,Precio,Ventas
6,2026-01-02,Laptop,Centro,Ana,2,15334.46,30668.92
11,2026-01-03,Laptop,Norte,Jorge,2,15435.73,30871.46
14,2026-01-04,Laptop,Centro,Ana,1,15310.72,15310.72
19,2026-01-05,Laptop,Sur,Paula,1,14788.77,14788.77
20,2026-01-05,Laptop,Sur,Carlos,1,14809.52,14809.52


## Parte 4 — Estadísticos

In [14]:
print("Media:   ", df["Ventas"].mean())
print("Mediana: ", df["Ventas"].median())
print("Mínimo:  ", df["Ventas"].min())
print("Máximo:  ", df["Ventas"].max())
print("Suma:    ", df["Ventas"].sum())
print("Conteo:  ", df["Ventas"].count())

Media:    7462.978758620689
Mediana:  3342.84
Mínimo:   426.28
Máximo:   48336.93
Suma:     7574923.4399999995
Conteo:   1015


In [15]:
# Apoyo para las reflexiones: cuánto aporta cada producto y cada región al total
por_producto = df.groupby("Producto")["Ventas"].agg(["count", "sum"])
por_producto["% del total"] = (por_producto["sum"] / df["Ventas"].sum() * 100).round(1)
print(por_producto.sort_values("sum", ascending=False))
print((df.groupby("Region")["Ventas"].sum() / df["Ventas"].sum() * 100).round(1))

           count         sum  % del total
Producto                                 
Laptop       182  4499637.28         59.4
Monitor      202  1374720.11         18.1
Impresora     69   615386.88          8.1
Teclado      198   513325.25          6.8
Mouse        219   299178.49          3.9
Audifonos    145   272675.43          3.6
Region
Centro    34.7
Este      15.2
Norte     26.0
Sur       24.1
Name: Ventas, dtype: float64


## Preguntas de la Parte 2

1. **¿Cuántas observaciones existen?** 1,015 filas, según `df.shape`.
2. **¿Cuántas variables existen?** 7 columnas.
3. **¿Qué representa una fila?** Una transacción de venta: un vendedor vendió cierta cantidad de un producto, a cierto precio, en una región y en una fecha.
4. **¿Qué representa cada columna?** `Fecha` es el día de la venta, `Producto` el artículo vendido, `Region` la zona, `Vendedor` quién la hizo, `Cantidad` las piezas vendidas, `Precio` el precio de cada pieza y `Ventas` el monto total de la transacción (Cantidad por Precio, como se ve en las primeras filas).
5. **Variables numéricas:** `Cantidad`, `Precio` y `Ventas`.
6. **Variables categóricas:** `Producto`, `Region` y `Vendedor`. `Fecha` también salió como texto (`str`), aunque representa una fecha; habría que convertirla con `pd.to_datetime` para usarla como tal.

## Preguntas de reflexión

1. **¿Qué variable consideras más importante del dataset?** `Ventas`, porque resume el resultado de cada transacción y es la que se usa para casi cualquier pregunta de negocio (qué producto, región o vendedor genera más). Para explicar por qué unas ventas son más grandes que otras, la clave es `Producto`, porque el precio depende casi por completo del tipo de producto.

2. **¿Qué información encontraste que no era evidente únicamente observando las primeras filas?** Que el dinero está muy concentrado en un solo producto. En `head()` se ven Mouse, Teclado y Audífonos con montos pequeños (entre $800 y $2,250), pero al agrupar resulta que las Laptop (182 de 1,015 transacciones) aportan el 59.4% de todas las ventas, mientras que Mouse y Audífonos juntos aportan solo 7.5%. También se nota que la media de `Ventas` ($7,463) es más del doble de la mediana ($3,343): unas pocas ventas grandes jalan el promedio hacia arriba, y 220 ventas (22% de las filas) superan los $10,000.

3. **¿Qué pregunta de negocio o investigación podría responder este dataset?** Por ejemplo, qué tan dependiente es la tienda de las laptops y qué pasaría si baja su demanda. También, qué región genera más ingresos (el Centro parece liderar con 34.7%) o si el desempeño de los vendedores depende de qué producto venden.

## Conclusiones

1. El dataset tiene 1,015 transacciones y 7 columnas, sin valores nulos, y va del 1 de enero al 10 de agosto de 2026. No se ve nada que obligue a limpiarlo antes de explorarlo.

2. Los productos tienen escalas de precio muy distintas (el Precio va de $418 a $16,112), así que las estadísticas globales de `Ventas` engañan un poco: la media ($7,463) es más del doble de la mediana ($3,343). Para describir una venta típica conviene usar la mediana.

3. Las ventas dependen mucho de un producto y de una región: las Laptop hacen el 59.4% del monto con solo 182 transacciones, y el Centro concentra el 34.7% frente al 15.2% del Este. Son hallazgos iniciales de la exploración; para confirmar diferencias entre regiones o vendedores harían falta análisis más formales.